### PR TITLE
Hvu 128

### PR DESCRIPTION
**Pull request by rld0528**

### CODE REVIEW

File: src/app/routes/secrets/SecretsList.js [hunk 1]
```diff
@@ -108,30 +108,35 @@ class SecretsList extends Component {
...
+                    this._openApprovedRequestModal(mount, name, name);
+                } else if (isWrapped) {
                     this._openApprovedRequestModal(mount, secretsPath, name);
                 } else {
                     this._toggleCreateUpdateSecretModal(`${mount}/${secretsPath}`, 'read');
                     openApprovedSecret(name, this._getVersionFromMount(mount));
                 }
             } else if (requiresRequest) {
-                const requestType = isWrapped ? CONTROL_GROUP : STANDARD_REQUEST;
+                const requestType = isDynamicSecret ? DYNAMIC_REQUEST : isWrapped ? CONTROL_GROUP : STANDARD_REQUEST;
```
Comment (jzhobes) [inline src/app/routes/secrets/SecretsList.js]: No nested ternary statements please. It gets kind of confusing.

File: src/app/routes/secrets/SecretsList.js [hunk 2]
```diff
@@ -461,7 +466,7 @@ class SecretsList extends Component {
             </Tooltip>;
         } else if (requiresRequest) {
             const requestAccessLabel = isPending ? 'Cancel Request' : 'Request Access';
-            const requestType = isWrapped ? CONTROL_GROUP : STANDARD_REQUEST;
+            const requestType = isDynamicSecret ? DYNAMIC_REQUEST : isWrapped ? CONTROL_GROUP : STANDARD_REQUEST;
```
Comment (jzhobes) [inline src/app/routes/secrets/SecretsList.js]: Same as previous comment. See `NotificationsModal._getRequestTypeLabel`. In this case, a simple if/else provides better clarity.

File: src/app/routes/secrets/SecretsList.js [hunk 3]
```diff
@@ -523,11 +527,11 @@ class SecretsList extends Component {
         } else if (secretsList.length > 0) {
             return <List>{
                 secretsList.map((secret, i) => {
-                    const {authorizationsText, name, secondaryText, url} = secret;
+                    const {authorizationsText, isDynamicSecret, name, secondaryText, url} = secret;
                     return <ListItem button component={this._renderLink} key={`key-${i}`} {...{onClick: this._onClickSecret.bind(this, secret), to: url}}>
                         <ListItemAvatar>
                             <Avatar>{
-                                name.endsWith('/') ? <FolderIcon/> : <FileCopyIcon/>
+                                isDynamicSecret ? <Cloud/> : name.endsWith('/') ? <FolderIcon/> : <FileCopyIcon/>
```
Comment (jzhobes) [inline src/app/routes/secrets/SecretsList.js]: New `_getSecretsIcon` method would be clearer.

File: src/app/routes/secrets/SecretsList.js [hunk 4]
```diff
@@ -683,90 +652,55 @@ const _mapStateToProps = (state, ownProps) => {
...
+        if (requiresRequest) {
+            secondaryText = `Request type: ${isWrapped ? 'Control Groups' : isDynamicSecret ? 'Dynamic Request' : 'Standard Request'}`;
+            // Check for any active requests.
+            const secretsPath = `${mount}${isV2 ? '/data/' : '/'}${currentPath}`;
+            activeRequest = secretsRequests.find((request) => request.path === secretsPath);
+            if (activeRequest) {
+                const {approved, authorizations, creationTime, opened} = activeRequest;
                 isOpened = opened;
+                isDSCanOpen = isDynamicSecret ? !isOpened : true;
+                isApproved = approved && isDSCanOpen;
```
Comment (jzhobes) [inline src/app/routes/secrets/SecretsList.js]: Is this correct? What about approved Control Group request?

Comment (rld0528) [reply src/app/routes/secrets/SecretsList.js]: if its control groups / standard request that will render it to true. In this part `isDynamicAndCanOpen = isDynamicSecret ? !isOpened : true;`

**Pull request merged**